docs/source/trainer/using_the_trainer.rst

```diff
@@ -461,6 +461,32 @@ points during training and (2) load them back to resume training later.
 
     The :doc:`checkpointing` guide.
 
+Gradient Accumulation
+~~~~~~~~~~~~~~~~~~~~~
+
+Composer supports gradient accumulation, which allows training arbitrary
+logical batch sizes on any hardware by breaking the batch into ``grad_accum``
+different microbatches.
+
+.. code:: python
+
+    from composer import Trainer
+
+    trainer = Trainer(
+        ...,
+        grad_accum=2,
+    )
+
+If ``grad_accum=auto``, Composer will try to automatically determine the 
+smallest ``grad_accum`` which the current hardware supports. In order to support automatic
+gradient accumulation, Composer initially sets ``grad_accum=1``. During the training process,
+if a Cuda Out of Memory Exception is encountered, indicating the current batch size is too
+large for the hardware, Composer catches this exception and continues training after doubling
+``grad_accum``. As a secondary benefit, automatic gradient accumulation is able to dynamically
+adjust throughout the training process. For example, when using ``ProgressiveResizing``, input
+size increases throughout training. Composer automatically increases ``grad_accum`` only when
+required, allowing for faster training at the start until image sizes are scaled up.
```

Comment:
Should add a few sentences on when it exists (e.g. if CUDA OOM , and grad_accum == batch_size).

Also should note that this is experimental!